au4/src/au3wrap/internal/domconverter.cpp

```diff
@@ -11,23 +12,22 @@ au::trackedit::TrackId DomConverter::trackId(const TrackId& au3trackId)
     return *(reinterpret_cast<const long*>(&au3trackId));
 }
 
-au::trackedit::Clip DomConverter::clip(const WaveTrack* waveTrack, const WaveClip* au3clip, int index)
+au::trackedit::Clip DomConverter::clip(const WaveTrack* waveTrack, const WaveClip* au3clip)
 {
-    //! TODO AU4 Just for tests
-    static std::vector<muse::draw::Color> colors = {
-        muse::draw::Color::fromString("#677CE4"),
-        muse::draw::Color::fromString("#2C79B0"),
-        muse::draw::Color::fromString("#2D9696")
-    };
-
     au::trackedit::Clip clip;
     clip.key.trackId = trackId(waveTrack->GetId());
-    clip.key.index = index;
+    clip.key.clipId = WaveClipID(au3clip).id;
     clip.title = wxToSting(au3clip->GetName());
     clip.startTime = au3clip->GetPlayStartTime();
     clip.endTime = au3clip->GetPlayEndTime();
 
-    size_t colorIdx = clip.key.trackId + clip.key.index;
+    //! TODO AU4 Just for tests
+    static std::vector<muse::draw::Color> colors = {
+        muse::draw::Color::fromString("#677CE4"),
+        muse::draw::Color::fromString("#2C79B0"),
+        muse::draw::Color::fromString("#2D9696")
+    };
+    size_t colorIdx = std::llabs(clip.key.trackId + clip.key.clipId);
```

Comment:
This would work as long as color count is not a multiple of 4, otherwise all clips would have the same color. But as I understand this is a temporary solution, so I believe this is fine for now.

https://godbolt.org/z/6Pz5jhWdb